backend/src/main/java/com/redhat/cloud/notifications/processors/email/RecipientResolver.java

```diff
@@ -0,0 +1,34 @@
+package com.redhat.cloud.notifications.processors.email;
+
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.recipients.User;
+import com.redhat.cloud.notifications.recipients.rbac.RbacRecipientUsersProvider;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class RecipientResolver {
+
+    @Inject
+    RbacRecipientUsersProvider rbacRecipientUsersProvider;
+
+    public Uni<List<User>> recipientUsers(String accountId, List<Endpoint> endpoints, Set<String> subscribers) {
+        return Multi.createFrom().iterable(endpoints)
+                .onItem().transformToUni(e -> recipientUsers(accountId, e, subscribers))
```

Comment:
Unless I misunderstood the code, something's wrong here:
- the endpoint is not actually used by `recipientUsers`
- the whole stream invokes `recipientUsers` n times if there are n endpoints
- each `recipientUsers` invocation returns the same list of values
- the resulting lists are flattened into a single list with duplicates removal, but the result is the same as any `recipientUsers` invocation

Reply:
> * the endpoint is not actually used by `recipientUsers`
> * the whole stream invokes `recipientUsers` n times if there are n endpoints
> * each `recipientUsers` invocation returns the same list of values
> * the resulting lists are flattened into a single list with duplicates removal, but the result is the same as any `recipientUsers` invocation


This is preparation for a follow-up PR, this will be it's next form: https://github.com/RedHatInsights/notifications-backend/pull/396/files#diff-8df1f99650e663bc8ff48a385fec4f0f1511608d2bdddaf457b373db1df6521dR32-R48

![image](https://user-images.githubusercontent.com/3845764/126209332-81d6ea8e-6061-449f-98a4-da10d94e944d.png)

Reply:
Also, endpoints are deduplicated at some point (or they should be - will double check). Right now we only have one EmailSubscription Endpoint, so `endpoints.size()` will always be at most 1.

Reply:
Oh that makes more sense with the next form 😄

Reply:
> edit: I could change the List to a Set to make it more obvious.

Which one, the list at line 25? It would be great I see that your follow-up PR already use an intermediate Set, I like that version better 😃

Reply:
I already changed the `List<Endpoint>` to `Set<Endpoint>`. Can do the same for line 25